lib/Wikia/tests/Service/User/UserPreferencesIntegrationTest.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Wikia\Service\User\Preferences;
+
+use Doctrine\Common\Cache\CacheProvider;
+use Wikia\DependencyInjection\Injector;
+use Wikia\Persistence\User\Preferences\PreferencePersistence;
+use PHPUnit_Framework_TestCase;
+
+class UserPreferencesIntegrationTest extends PHPUnit_Framework_TestCase {
+	/**
+	 * @var string
+	 */
+	protected $testUser;
+
+	/**
+	 * @var int
+	 */
+	protected $testUserId;
+
+	/**
+	 * @var PreferenceService
+	 */
+	protected $preferenceServiceCached;
+
+	/**
+	 * @var PreferenceService
+	 */
+	protected $preferenceServiceUnCached;
+
+	/**
+	 * @var string
+	 */
+	const TEST_PREFERENCE_NAME = "hidepatrolled";
+
+	/**
+	 * @param bool $ignoreCache should we return not cached PreferenceService instance
+	 * @return PreferenceService
+	 */
+	private function getUserPreferences($ignoreCache = false) {
+		if ( $ignoreCache ) {
+			return $this->preferenceServiceUnCached;
+		} else {
+			return $this->preferenceServiceCached;
+		}
+	}
+
+	function setUp() {
+		global $wgWikiaBotUsers;
+		$this->testUser = $wgWikiaBotUsers["bot"];
+		$this->testUserId = \User::idFromName( $this->testUser );
+		$this->preferenceServiceCached = Injector::getInjector()->get( PreferenceService::class );
+		$defaultPreferences = Injector::getInjector()->get( PreferenceServiceImpl::DEFAULT_PREFERENCES );
+		$cache = $this->getMockBuilder( CacheProvider::class )
```

Comment:
There's also `VoidCache` which doesn't cache anything. Don't *need* to change, but in the future it'd be nice to use that since it was made specifically for tests